src/mtcp/mtcp_restart.c

```diff
@@ -970,14 +975,20 @@ static void read_shared_memory_area_from_file(int fd, Area* area, int flags)
 
   /* Check to see if the filename ends with " (deleted)" */
   if (mtcp_strendswith(area_name, DELETED_FILE_SUFFIX)) {
+    // FIXME:  Shouldn't we also unlink area_name after re-mapping it?
     area_name[ mtcp_strlen(area_name) - mtcp_strlen(DELETED_FILE_SUFFIX) ] =
       '\0';
+    if (fix_parent_directories(area_name) != area_name) {
```

Comment:
Since we are already calling `fix_parent_directories` above, why do we need it here as well?